src/nemos/io/io.py

```diff
@@ -0,0 +1,228 @@
+    """
+    Load a previously saved nemos model from a .npz file.
+
+    This will read the model parameters from the specified file and instantiate
+    the model class with those parameters. It allows for custom mapping of
+    attribute names to their actual objects using a mapping dictionary.
+
+    Parameters
+    ----------
+    filename :
+        Path to the saved .npz file.
+    mapping_dict :
+        Optional dictionary to map custom attribute names to their actual objects.
+
+    Returns
+    -------
+    model :
+        An instance of the model class with the loaded parameters.
+
+    Examples
+    --------
+    >>> import nemos as nmo
+    >>> import jax.numpy as jnp
+    >>> # Create a GLM model with specified parameters
+    >>> solver_args = {"stepsize": 0.1, "maxiter": 1000, "tol": 1e-6}
+    >>> model = nmo.glm.GLM(
+    ...     regularizer="Ridge",
+    ...     regularizer_strength=0.1,
+    ...     observation_model="Gamma",
+    ...     solver_name="BFGS",
+    ...     solver_kwargs=solver_args,
+    ... )
+    >>> for key, value in model.get_params().items():
+    ...     print(f"{key}: {value}")
+    observation_model__inverse_link_function: <function one_over_x at ...>
+    observation_model: GammaObservations(inverse_link_function=one_over_x)
+    regularizer: Ridge()
+    regularizer_strength: 0.1
+    solver_kwargs: {'stepsize': 0.1, 'maxiter': 1000, 'tol': 1e-06}
+    solver_name: BFGS
+    >>> # Save the model parameters to a file
+    >>> model.save_params("model_params.npz")
+    >>> # Create a custom link function
+    >>> def custom_link_function(x):
+    ...     return jnp.pow(x, 2)
+    >>> observation_model = nmo.observation_models.PoissonObservations(
+    ...     inverse_link_function=custom_link_function
+    ... )
+    >>> mapping_dict = {
+    ...     "observation_model": observation_model,
+    ...     "regularizer": "Lasso",
+    ...     "solver_name": "ProxSVRG",
+    ... }
+    >>> # Load the model from the saved file with custom mapping
+    >>> model = nmo.load_model("model_params.npz", mapping_dict=mapping_dict)
+    >>> # The model is now loaded, overwriting the observation model and regularizer
+    >>> for key, value in model.get_params().items():
+    ...     print(f"{key}: {value}")
+    observation_model__inverse_link_function: <function custom_link_function at ...>
+    observation_model: PoissonObservations(inverse_link_function=custom_link_function)
+    regularizer: Lasso()
+    regularizer_strength: 0.1
+    solver_kwargs: {'stepsize': 0.1, 'maxiter': 1000, 'tol': 1e-06}
+    solver_name: ProxSVRG
+    >>> # Test the custom link function does what we expect
+    >>> x = jnp.array([5.0])
+    >>> expected_output = jnp.array([25.0])
+    >>> output = model.observation_model.inverse_link_function(x)
+    >>> print(f"Custom link function output: {output}, Expected: {expected_output}")
+    Custom link function output: [25.], Expected: [25.]
```

Comment:
```suggestion
    >>> model.observation_model.inverse_link_function(x)
    [25.]
```

this does the same thing (testing that the custom link function has the expected output) and is the more standard way to do so in doctests